includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -76,7 +76,6 @@ class WC_Order_Data_Store_CPT extends Abstract_WC_Order_Data_Store_CPT implement
 	public function create( &$order ) {
 		$order->set_order_key( 'wc_' . apply_filters( 'woocommerce_generate_order_key', uniqid( 'order_' ) ) );
 		parent::create( $order );
-		do_action( 'woocommerce_new_order', $order->get_id() );
```

Comment:
> `woocommerce_new_order` still fires in the same way, just in additional places like API.

IMO changing the context of a hook is a breaking change. Especially knowing that different contexts won't have the same globals available, like `WC()->cart`. This plugin breaking with a new version of WC and no changes to its own code is evidence of that.

I don't think it's a big issue for 3rd party code to have to check for global dependencies so I agree it's best not to merge this PR. It's just something that should be considered a breaking change and documented as such (or better yet, avoided).

> this action is important for both too.

I agree _an_ action is important, but not _this_ action. The hook doesn't need to be an existing hook previously triggered only under the checkout context and then repurposed to be triggered in new contexts. That's the breaking change.

A safer approach would have been to add a new hook for the new general case (e.g. `woocommerce_order_created`). The old `woocommerce_new_order` hook could still be triggered with no change to context and a deprecated notice. That maintains backward compatibility while also providing the benefit of a unified hook for all contexts.